fdbserver/VersionedBTree.actor.cpp

```diff
@@ -7308,15 +7476,24 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 		Version remapCleanupWindow =
 		    (BUGGIFY ? deterministicRandom()->randomInt64(0, 100) : SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_WINDOW);
 
+		EncodingType encodingType = EncodingType::XXHash64;
+
+		// Deterministically enable encryption based on uid
+		if (g_network->isSimulated() && logID.hash() % 2 == 0) {
```

Comment:
How about protecting this code under some Knob to control when it gets enabled? 
The plan so far is to enable encryption all the time, but, enabling it "selectively" is something that would be desirable for sure. 
Thanks

Reply:
This code selects XOR encryption randomly in simulation.  There is no need to disable that logic with a knob, it's better for all simulations to randomly test with and without encryption at this stage.  Eventually, 'real' encryption options will need to be plumbed in from Worker to this constructor and that will include the key provider interface.  At that point, we can move the randomly chosen XOR option to that level.

Reply:
In the end, the encryption mode will have to be decided in SimulatedCluster for simualtions, and persisted in the saved state for restarting tests.  For real users, the encryption mode probably has to be stored in the coordinated state for the cluster because a) tlogs need it to recover and b) a storage engine can't store it for obvious reasons.

Reply:
Agree; didn't think about _restartingTest_, you are correct it needs to be persisted (may be co-ordinated state). Will add it to my TODO list to make sure it gets addressed. 
Thanks